pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -435,6 +448,16 @@ func (r *ReconcileClusterDisruption) hasPGHealthCheckTimedout(pdbStateMap *corev
 	return false
 }
 
+// isNodeDrained returns true if a node is unschedulable
+func isNodeDrained(client client.Client, nodeName string) (bool, error) {
+	node := &corev1.Node{}
+	if err := client.Get(context.TODO(), types.NamespacedName{Name: nodeName}, node); err != nil {
```

Comment:
Is the `labels[osdHostNameLabel]` the immutable K8s node name, or is it the dns host name (that might change)? I believe it's the K8s node name (which I would expect), but just wanted to confirm.

Reply:
Thanks for pointing that out.  This label is from the `--crush-location=root=default host=minikube-m04`.  So it will work for OSD on disks but not for osd on pvc, where `host` in the `--crush-location` is the PVC name - ` --crush-location=root=default host=ocs-deviceset-gp2-0-data-0g6gw9 region=us-east-1 zone=us-east-1c`.  

So it fails on OSD on pvc. 
`2021-04-23 07:05:04.736715 E | clusterdisruption-controller: failed to check if node is drained: failed to get node instance with name "ocs-deviceset-gp2-1-data-0k8tq9": Node "ocs-deviceset-gp2-1-data-0k8tq9" not found` 

 Need to get the host name differently.

Reply:
Checked a bit more.  There is no information about the node on the osd deployment when the node is drained.

Reply:
Some other options: 
1 check  osd `deployment.Spec.Template.Spec.NodeName`.  Empty value would mean that pod is not assigned to any node. And we can assume that node is draining. 
2. Get all the rook nodes in the cluster and check if any of the node has `node.Spec.Unschedulable` as `true`

Reply:
Yes, that sounds like a good approach. To add onto that, what about the following? 
1. The OSD is `down` according to ceph 
AND 
2. The `deployment.Spec.Template.Spec.NodeName` is empty (the pod is likely in Pending state waiting for a node to be available) OR the `deployment.Spec.Template.Spec.NodeName` is set and the node is `Unschedulable` (the pod is likely in Terminating state)

Reply:
>>> The OSD is down according to ceph
Ceph might take some time to report OSD down.